### PR TITLE
Remove references to wazuh.yml file and adapt to the new configuration system

### PR DESCRIPTION
**Pull request by Desvelao**
## Description

This pull request removes the references to the configuration of the Wazuh plugins using the `wazuh.yml` file and adapt to the new system.

Changes:
- Remove references to wazuh.yml
- Add documentation about the new system
- Add new Wazuh dashboard settings:
  - `wazuh_core.configuration.encryption_key`
  - `wazuh_core.instance`

### Docs building
- [ ] Compiles without warnings.
### Code formatting and web optimization
- [ ] Uses three spaces indentation.
- [ ] Adds or updates meta descriptions accordingly.
- [ ] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [ ] Uses present tense, active voice, and semi-formal registry.
- [ ] Uses short, simple sentences.
- [ ] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

**Review by asteriscos — CHANGES_REQUESTED** [review]
I found several references to the `wazuh.yml` file. We must analyze each of the following references to check if they must be deleted or changed:

https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/amazon-machine-images/amazon-machine-images.rst?plain=1#L92-L102


https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/docker/upgrading-wazuh-docker.rst?plain=1#L76-L83

https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/docker/wazuh-container.rst?plain=1#L431-L437

https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/virtual-machine/virtual-machine.rst?plain=1#L110-L114

https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/migration-guide/wazuh-dashboard.rst?plain=1#L122-L124

https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/user-manual/wazuh-dashboard/troubleshooting.rst?plain=1#L78-L81

**Review by javimed — COMMENTED** [review]
We could add a screenshot of **Dashboard management** > **Server APIs** highlighting the **Run as** field in the same way we currently have a `wazuh.yml` sample highlighting the `run_as` value. It would be placed at the same location as the yaml sample (between "If Run as is set to false" and "If Run as is set to true" repeating on many of the files.

Same goes for other wazuh.yml samples this PR is removing. We could better replace them with a shot of the configuration screen.

File: source/user-manual/wazuh-dashboard/config-file.rst [hunk 1]
```diff
@@ -1,18 +1,72 @@
 .. Copyright (C) 2015, Wazuh, Inc.
 
 .. meta::
-  :description: The Wazuh dashboard includes a configuration file where you can define custom values for several options. Learn more about it in this section.
+  :description: The Wazuh dashboard includes a configuration where you can define custom values for several options. Learn more about it in this section.
```
Comment (javimed) [inline source/user-manual/wazuh-dashboard/config-file.rst:4]:
```suggestion
   :description: The Wazuh dashboard includes configuration options to define custom settings. Learn more about it in this section.
```

File: source/user-manual/wazuh-dashboard/config-file.rst [hunk 2]
```diff
@@ -1,18 +1,72 @@
...
 
 .. _wazuh_dashboard_config_file:
 
-Configuration file
-==================
+Configuration
+=============
+
+The configuration related to Wazuh plugins can be defined in:
```
Comment (javimed) [inline source/user-manual/wazuh-dashboard/config-file.rst:11]:
```suggestion
You can configure the Wazuh dashboard and plugins as follows:
```

Comment (javimed) [reply source/user-manual/wazuh-dashboard/config-file.rst:11]: We use active voice as much as possible

File: source/user-manual/wazuh-dashboard/config-file.rst [hunk 3]
```diff
@@ -1,18 +1,72 @@
...
+
+- **Plugins settings**:
```
Comment (javimed) [inline source/user-manual/wazuh-dashboard/config-file.rst:13]: -  **Wazuh dashboard settings**: You can edit the Wazuh dashboard configuration file.

   -  ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` for installations from packages.
   -  ``/usr/share/wazuh-dashboard/config/opensearch_dashoards.yml`` for Docker installations.

-  **Plugins settings**: Users with privileges to manage the Wazuh indexer REST API can manage the plugins settings from the Wazuh dashboard. Wazuh stores the custom values in a saved object.

   -  **Dashboard management** > **App Settings**: To customize several settings.
   -  **Dashboard management** > **Server APIs**: To customize API host entries.

**Review by javimed — CHANGES_REQUESTED** [review]
Sections hierarchy isn't enough clear

imagen

File: source/user-manual/wazuh-dashboard/config-file.rst (hunk 3 above)
Comment (Desvelao) [reply source/user-manual/wazuh-dashboard/config-file.rst]: Applied the suggestion on: 7211ef37558ea73c7b1dd235ed7dfe7e2e23f6bb

**Comment by Desvelao** [comment]
> I found several references to the `wazuh.yml` file. We must analyze each of the following references to check if they must be deleted or changed:
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/amazon-machine-images/amazon-machine-images.rst?plain=1#L92-L102
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/docker/upgrading-wazuh-docker.rst?plain=1#L76-L83
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/docker/wazuh-container.rst?plain=1#L431-L437
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/virtual-machine/virtual-machine.rst?plain=1#L110-L114
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/migration-guide/wazuh-dashboard.rst?plain=1#L122-L124
> 
> https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/user-manual/wazuh-dashboard/troubleshooting.rst?plain=1#L78-L81

I removed some references:
- https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/amazon-machine-images/amazon-machine-images.rst?plain=1#L92-L102
- https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/deployment-options/virtual-machine/virtual-machine.rst?plain=1#L110-L114
- https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/user-manual/wazuh-dashboard/troubleshooting.rst?plain=1#L78-L81
I decided to remove this entry in the troubleshooting because this error is not thrown anymore. See the details for more information.

<details>

This error was caused by the API endpoint `POST /api/check-api` that validated the request payload. but this validation was commented so does not apply: https://github.com/wazuh/wazuh-dashboard-plugins/commit/bbeea3ac45e7678427356842856e4d55bd6f0b7f
</details>

The references to the upgrade the Docker deployment depend on the Wazuh dashboard image for that version. This has to be communicated to the responsible team.

Same topic with the reference in the migration guide from the Kibana to the Wazuh dashboard.
https://github.com/wazuh/wazuh-documentation/blob/66a0a03da063475d46dd190d4c523aa8a1d43a0a/source/migration-guide/wazuh-dashboard.rst?plain=1#L122-L124

**Comment by Desvelao** [comment]
> We could add a screenshot of **Dashboard management** > **Server APIs** highlighting the **Run as** field in the same way we currently have a `wazuh.yml` sample highlighting the `run_as` value. It would be placed at the same location as the yaml sample (between "If Run as is set to false" and "If Run as is set to true" repeating on many of the files.
> 
> Same goes for other wazuh.yml samples this PR is removing. We could better replace them with a shot of the configuration screen.

I added a screenshot to check if the `run_as` is enabled or not.

I will review the other references you mentioned.

**Comment by Desvelao** [comment]
I added more screenshots about the addition and edition of an API connection and checking the `Run as` status.

File: source/user-manual/user-administration/ldap.rst [hunk 4]
```diff
@@ -233,20 +233,10 @@ Follow these steps to create a new role mapping and grant administrator permissi
...
-      hosts:
-        - default:
-            url: https://localhost
-            port: 55000
-            username: wazuh-wui
-            password: "<wazuh-wui-password>"
-            run_as: false
-
-   If ``run_as`` is set to ``true``, you need to add a role mapping on the Wazuh dashboard. To map the backend role to Wazuh, follow these steps:
+   .. image:: /images/wazuh-dashboard/api-connections/wazuh-dashboard-check-api-connection-run-as.jpg
```
Comment (Desvelao) [inline source/user-manual/user-administration/ldap.rst:238]: I'm not sure which image is more appropriate to show here, if the `Run as` is disabled or enabled. What do you think about this @javimed ?

**Comment by Desvelao** [comment]
> Sections hierarchy isn't clear enough.
> 
> imagen

Do you have a proposal for an improvement @javimed ?

CC: @asteriscos

**Comment by javimed** [comment]
> Do you have a proposal for an improvement @javimed ?
> 
> CC: @asteriscos

The structure could reflect the alternatives and options the user has. If the configuration of the Wazuh dashboard is different to that of the plugins, then this whole Configuration section should have two clearly separated sub-sections:

- One for configuring the dashboard
- One for configuring the Plugins

The way it is now I'm not sure if `Setup a configuration` applies only to the plugins configuration. In addition I can see `General options` at the same level as `Configuration file` and I don't know what configuration file are we talking about here nor if this is useful to configure the Wazuh dashboard or the plugins. There's an `Example` subsection but it's not clear to me if this belongs to configuring the plugins or what. Then I can see a Configuration of the Wazuh dashboard section showing some options and it's not clear what file to edit to use them. And I even wonder if maybe there's some shared configuration for both dashboard and plugins preventing this clear separation I'm mentioning.

File: source/user-manual/user-administration/ldap.rst (hunk 4 above)
Comment (javimed) [reply source/user-manual/user-administration/ldap.rst:238]: Sorry I overlooked this question, @Desvelao . We could put the `Run as` is _disabled_ just like the code block we currently have showing `false`. The image would go after the "_If Run as is set to false..._" text but before the "_If Run as is set to true..._" text just like the current code block we have. I think run as false is the simplest configuration option and worth displaying.

Comment (Desvelao) [reply source/user-manual/user-administration/ldap.rst:238]: The displayed value as `Run as` in the API connections table is not the same value ( `false` / `true` ) that is configured in the API hosts. The raw value of the `Run as` setting is not displayed. Instead, the displayed value can display a missconfiguration.

CC: @asteriscos

File: source/user-manual/wazuh-dashboard/config-file.rst [hunk 5]
```diff
@@ -1,18 +1,69 @@
...
+
+   -  ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` for installations from packages.
+   -  ``/usr/share/wazuh-dashboard/config/opensearch_dashoards.yml`` for Docker installations.
+
+- **Plugins settings**: Users with privileges to manage the Wazuh indexer security REST API can manage the plugins settings from the Wazuh dashboard. Wazuh stores the custom values in a saved object.
+
+   -  **Dashboard management** > **App Settings**: To customize several settings.
+   -  **Dashboard management** > **Server APIs**: To customize API host entries.
+
+This section describes all the settings available for each location.
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply source/user-manual/wazuh-dashboard/config-file.rst:23]: Plugins settings and Dashboard settings.

File: source/user-manual/wazuh-dashboard/config-file.rst [hunk 6]
```diff
@@ -1,18 +1,69 @@
...
+=============
+
+You can configure the Wazuh dashboard and plugins as follows:
+
+- **Wazuh dashboard settings**: You can edit the Wazuh dashboard configuration file.
+
+   -  ``/etc/wazuh-dashboard/opensearch_dashboards.yml`` for installations from packages.
+   -  ``/usr/share/wazuh-dashboard/config/opensearch_dashoards.yml`` for Docker installations.
+
+- **Plugins settings**: Users with privileges to manage the Wazuh indexer security REST API can manage the plugins settings from the Wazuh dashboard. Wazuh stores the custom values in a saved object.
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply source/user-manual/wazuh-dashboard/config-file.rst:18]: I applied this comment as a review suggestion: https://github.com/wazuh/wazuh-documentation/pull/7050/commits/cc7ec0b507bccdbaea494de04f3f0ed0d66fca04#r1523495979

**Comment by Desvelao** [comment]
I will close this pull request because the development changed as commented here: https://github.com/wazuh/wazuh-documentation/pull/7050#issuecomment-2009404648.

We did roll back on this issue https://github.com/wazuh/wazuh-dashboard-plugins/issues/6570 to some features added on this pull request .

**Pull request closed without merging**